WCFDataService/Client/System/Data/Services/Client/ClientEdmModel.cs

```diff
@@ -443,7 +447,8 @@ private EdmTypeCacheValue GetOrCreateEdmTypeInternal(IEdmStructuredType edmBaseT
                             List<IEdmProperty> loadedProperties = new List<IEdmProperty>();
                             foreach (PropertyInfo property in ClientTypeUtil.GetPropertiesOnType(type, /*declaredOnly*/edmBaseType != null).OrderBy(p => p.Name))
                             {
-                                IEdmProperty edmProperty = this.CreateEdmProperty(complexType, property);
+                                FieldInfo backingField = FindBackingField(property, this.ProxyBackingFieldNamingConvention);
```

Comment:
For performance reasons I think we should avoid accidentally instantiating complex properties as well. You could have entities with large and/or many complex properties that may not be included in some projected Query - it would be wrong to instantiate those complex properties.

As for properties on the complex types, I will omit backing field detection on those as they will never include navigation properties.